src/ircode.c

```diff
@@ -801,7 +795,9 @@ JL_DLLEXPORT jl_string_t *jl_compress_ir(jl_method_t *m, jl_code_info_t *code)
 
     if (m->roots == NULL) {
         m->roots = jl_alloc_vec_any(0);
+        m->roots_table = jl_alloc_memory_any(0);
         jl_gc_wb(m, m->roots);
+        jl_gc_wb(m, m->roots_table);
```

Comment:
```suggestion
        m->roots = jl_alloc_vec_any(0);
        jl_gc_wb(m, m->roots);
        m->roots_table = jl_alloc_memory_any(0);
        jl_gc_wb(m, m->roots_table);
```

I think the order here is important since otherwise GC can run during `jl_alloc_memory_any` and then miss the new `jl_alloc_vec_any`